packages/firebase_storage/android/src/main/java/io/flutter/plugins/firebase/storage/FirebaseStoragePlugin.java

```diff
@@ -46,12 +46,36 @@ public void onMethodCall(MethodCall call, final Result result) {
       case "StorageReference#getData":
         getData(call, result);
         break;
+      case "StorageReference#deleteFile":
+        deleteData(call, result);
+        break;
       default:
         result.notImplemented();
         break;
     }
   }
 
+  private void deleteData(MethodCall call, final Result result) {
+    Map<String, String> arguments = (Map<String, String>) call.arguments;
+    String path = arguments.get("path");
+    StorageReference ref = firebaseStorage.getReference().child(path);
+    final Task<Void> deleteTask = ref.delete();
+    deleteTask.addOnSuccessListener(
+        new OnSuccessListener<Void>() {
+          @Override
+          public void onSuccess(Void aVoid) {
+            result.success(true);
+          }
+        });
+    deleteTask.addOnFailureListener(
+        new OnFailureListener() {
+          @Override
+          public void onFailure(@NonNull Exception exception) {
+            result.error("deletion_error", exception.getMessage(), false);
```

Comment:
Seems like the error details should be null here, `false` is indicating that the method call failed, but the fact that an exception is being passed back to the Dart side would indicate that. So no need to pass in false here.